File: solutions/alfi.compitest/source_gen/alfi/compitest/Cars/Car.cpp
#include "Car.h"
#include <alf/library/BasicInputOutput.hpp>
#include <string>
#include "Vehicle.h"
#include <vector>
#include <memory>

namespace Cars
{
  int Car::getSpeed()
  {
    return 10;
  }
  void Car::startEngine()
  {
    alf::library::BasicInputOutput::WriteLine(std::string("Car engine started"));
    alf::library::BasicInputOutput::WriteLine(std::string("Initial speed: ") + std::to_string(this->getSpeed()));

    Cars::Vehicle::defaultStartEngine();
  }
  std::string Car::getColor()
  {
    return this->color;
  }
  void Car::setColor(std::string color)
  {
    this->color = color;
  }
  void Car::addWheel(std::shared_ptr<Car::Wheel> wheel)
  {
    alf::library::BasicInputOutput::WriteLine(std::string("Adding wheel: ") + std::to_string(wheel->size));
    this->wheels.push_back(wheel);
  }
}

File: solutions/alfi.compitest/source_gen/alfi/compitest/Cars/Car.cs
/// <filename>
///     Car.cs
/// </filename>
using Alf.Library;
using System;
using Cars;
using System.Collections.Generic;

namespace Cars
{
    public class Car : Cars.Vehicle, Cars.IHasColor
    {
        protected override int getSpeed()
        {
            return 10;
        }

        public override void startEngine()
        {
            BasicInputOutput.WriteLine("Car engine started");
            BasicInputOutput.WriteLine("Initial speed: " + Convert.ToString(this.getSpeed()));

            base.defaultStartEngine();
        }

        public virtual string getColor()
        {
            return this.color;
        }

        public virtual void setColor(string color)
        {
            this.color = color;
        }

        private string color = "Black";

        public class Wheel
        {
            public int size = 10;
        }

        public System.Collections.Generic.List<Cars.Car.Wheel> wheels = new System.Collections.Generic.List<Cars.Car.Wheel>();

        public virtual void addWheel(Cars.Car.Wheel wheel)
        {
            BasicInputOutput.WriteLine("Adding wheel: " + Convert.ToString(wheel.size));
            this.wheels.Add(wheel);
        }

        public enum ModelKind
        {
            PKW /* no value */,
            Limousine /* no value */
        }

        public struct Model
        {
            public string Name;

            public Cars.Car.ModelKind Kind;

            public Model()
            {
            }

            public Model(string Name, Cars.Car.ModelKind Kind)
            {
                this.Name = Name;
                this.Kind = Kind;
            }
        }

        public Cars.Car.Model model;
    }
}
